### ISSUE TITLE
Binary release of 1.3.4 does not work on Ubuntu 18.04

### ISSUE THREAD

**Issue by nlburgin** [clementgallet/libTAS#216]
Running libTAS 1.3.4 from either the .deb package or from the binary tar.gz produces this output:

```
Interim commit 9170d91 built on 2019-04-08
qt5ct: using qt5ct plugin
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QComboBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QComboBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QTableWidget
QObject::connect: signal not found in QTableWidget
QObject::connect: signal not found in QTableWidget
QObject::connect: signal not found in QTableWidget
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QSpinBox
QObject::connect: signal not found in QSlider
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QCheckBox
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QHeaderView
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in InputEditorView
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QDialogButtonBox
QObject::connect: signal not found in QActionGroup
QObject::connect: signal not found in QActionGroup
QObject::connect: signal not found in QActionGroup
QObject::connect: signal not found in QActionGroup
QObject::connect: signal not found in QActionGroup
QObject::connect: signal not found in QActionGroup
QObject::connect: signal not found in QActionGroup
QObject::connect: signal not found in QActionGroup
qt5ct: D-Bus global menu: no
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QAction
QObject::connect: signal not found in QComboBox
QObject::disconnect: signal not found in QComboBox
QObject::connect: signal not found in QComboBox
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QLineEdit
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QRadioButton
QObject::connect: signal not found in QRadioButton
QObject::connect: signal not found in QCheckBox
QObject::connect: signal not found in QCheckBox
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QPushButton
QObject::connect: signal not found in QGroupBox
QObject::disconnect: signal not found in QComboBox
QObject::connect: signal not found in QComboBox
```

And then the GUI comes up, but none of the buttons work.

It doesn't do that if I build it myself. Where are these binaries getting built? Are they getting built on your machine? 

If so, I'm guessing it might be related to #212. You fixed the build error when building with older versions of Qt using a preprocessor macro omitting references to new features. Even still, if you build it *with* those features (which would probably happen on your machine) and then distribute that binary to an older system, it's liable to malfunction.

Assuming that assumption is correct, here's all the possible fixes I can think of:

**1.** Only build the publicly released binaries against the oldest supported version of Qt.
**2.** Bundle libTAS with the Qt libraries it was built against so it uses those instead of depending on the system's installed version.
**3.** The second option, but take it a step further and actually static-link the Qt libraries *into* LibTAS instead of just bundling the relevant .so files.
**4.** Some "extreme" form of environment bundling such as an AppImage.

Each option has its own pros and cons.

**Comment by clementgallet** [comment]
I'm building the release version on VMs with Debian Stretch (stable), which contains an old version of Qt (5.7). The issue is only present on the debian package build, I'm not getting any issue when compiling it myself on the same machine, so it may be related to #213

**Comment by clementgallet** [comment]
I re-enabled hardening, which fixed the issue (54098e9952db7ca7709411e7be946e0de8bb1598). I deleted the release and made a new one with the updated debian packages

**Comment by nlburgin** [comment]
Oops. Sorry about that...

Yeah, the new package works now.
